plc4j/spi/src/main/java/org/apache/plc4x/java/spi/connection/SingleProtocolStackConfigurer.java

```diff
@@ -42,69 +47,57 @@
 public class SingleProtocolStackConfigurer<BASE_PACKET_CLASS extends Message> implements ProtocolStackConfigurer<BASE_PACKET_CLASS> {
```

Comment:
The whole idea of the `SingleProtocolStackConfigurer ` is to work class based.
So maybe we rename this configurer into `SingleDIProtocolStackConfigurer` and we create a second `SingleSupplierProtocolStackConfigurer`.
Like now you force every stack to use the new api instead of providing an alternate for for drivers where its neccessary.

Reply:
I have removed configuration handling from configurer cause now it is available to builder. Given above we now can have multiple builders which construct stack in different ways. One note, leaving `SingleProtocolStackConfigurer` with its initial class based approach will require us to introduce second type which will differ only by single lines. These lines will be calls to suppliers instead of reflection.

Are you suggesting to revert `SingleSupplierProtocolStackConfigurer` and duplicate its functionality to other place?

Reply:
Basically yes: This also would clean up the PR diff as we could then clearly see at which point we need the second configurer. The missing documentation on the configurer is on me (we where pretty time limited back then) :)
I will document it ASAP.

Reply:
oh and btw.: I would recommend to let the supplier accept the configuration as an argument. This way you could always decided for yourself if you would work with the configuration or just ignore it.